### ISSUE TITLE
Avoinding negative K and Omega values in SST 

### ISSUE THREAD

**Issue by msv5450** [NaluCFD/Nalu#429]
in "ShearStressTransportEquationSystem -->  update_and_clip()"

I understand that you try to correct negative solved values of turbulent kinetic energy and dissipation rate because they are not physically sensible, but why do you set turbulent viscosity equal to the molecular viscosity if one of them is negative? I think turbulent viscosity should be zero in that case, similar to what is done in the Ksgs model if "k < 0".

Also, in "SpecificDissipationRateSSTNodeSourceSuppAlg.C", you've got:

**lhs[0] += (2.0*beta*rho*sdr + std::max(Sw / sdr,0.0))*dualVolume;**

I think "std::max(Sw / sdr,0.0)" should be replaced by "Sw * std::max(sdr,0.0)".

**Comment by msv5450** [comment]
Hi Stefan,

Thanks for your explanation.

My name is MJ Sarfi and I am a PhD student in Professor Greg Wagner's group at Northwestern university. I am familiar with CVFEM discretization of equations and how Nalu works in general. I have modified and used NALU so far for simulating additive manufacturing processes. My main interest in Nalu is simulation of fluid flow in the melt pool, which a region in the domain where the alloy gets melted by the laser power and forms a liquid phase. This leads to a complex velocity field that is driven by temperature gradient and surface tension forces. All my simulations were done with a laminar assumption so far.

Right now, I want to consider the effect of turbulence in the melt pool flow. I've seen several cases in the literature that including the k-epsilon turbulence model results in fairly accurate melt pool fluid flow predictions that are consistent with experiments. So, Greg asked me to add the k-epsilon model to the existing version of Nalu that we have in our group and test it. I've just finished adding k-epsilon and I used SST RANS model that is already available in Nalu src as a template for creating the k-epsilon equation system. I also added buoyancy source terms for K and epsilon equations because my research problem involves enthalpy as well.
